brainzutils/musicbrainz_db/utils.py

```diff
@@ -20,27 +20,33 @@ def get_entities_by_gids(query, entity_type, mbids):
         Dictionary of objects of target entities keyed by their MBID.
     """
     entity_model = ENTITY_MODELS[entity_type]
-    results = query.filter(entity_model.gid.in_(mbids)).all()
-    remaining_gids = list(set(mbids) - {entity.gid for entity in results})
-    entities = {str(entity.gid): entity for entity in results}
-
     if entity_type in META_MODELS:
         meta_model = META_MODELS[entity_type]
         query = query.add_entity(meta_model).join(meta_model)
-        entity_ids = list({entity.id for entity in results})
-        results = query.filter(meta_model.id.in_(entity_ids))
+
+    results = query.filter(entity_model.gid.in_(mbids)).all()
+    entity_gids = set()
+    entities = {}
+    if entity_type in META_MODELS:
```

Comment:
Unrelated to current PR but I am unsure why this is fetch ratings for found recordings first then check redirects and then fetch their ratings. Wouldn't the opposite, fetch recordings then redirects after that ratings?

Reply:
we discussed this - the new fixes means that only 2 queries are made. 1 to get the entity + meta, and a 2nd (if needed) to get redirect mbids and their ratings